packages/lit-dev-content/src/components/litdev-tutorial.ts

```diff
@@ -255,53 +438,53 @@ export class LitDevTutorial extends LitElement {
     project.projectSrc = src;
   }
 
-  private async _loadStep() {
-    this._loading = true;
-    this._solved = false;
-    const active = this._info;
-    this._html =
-      this._preloadedHtml?.idx === this._idx
-        ? await this._preloadedHtml.promise
-        : await this._fetchHtml(active.htmlSrc);
-    this._setProjectSrc(active.projectSrcBefore);
-    this._loading = false;
-    // Use scrollTop instead of scrollIntoView, because scrollIntoView also
-    // changes focus.
-    this.renderRoot.querySelector('#tutorialContent')!.scrollTop = 0;
-
-    // Start loading the next step's HTML content.
-    const next = this._nextInfo;
-    if (next !== undefined) {
-      this._preloadedHtml = {
-        idx: next.idx,
-        promise: this._fetchHtml(next.htmlSrc),
-      };
-    }
-  }
-
+  /**
+   * Fetches the text of a file. Typically expected to be the rendered HTML of
+   * the markdown tutorial step descriptions
+   *
+   * @param src URL to fetch
+   * @returns stringified text of the file at the given location
+   */
   private async _fetchHtml(src: string): Promise<string> {
     return (await fetch(src)).text();
   }
 
+  /**
+   * Finds the index of the given slug. e.g. "" -> 0, "03" -> 3
+   *
+   * @param slug slug of the step
+   * @returns The index with the given slug
+   */
   private _slugToIdx(slug: string): number | undefined {
-    const idx = manifest.steps.findIndex((step) => step.slug === slug);
-    return idx === -1 ? undefined : idx;
+    const idx = Number(slug);
+    return isNaN(idx) ? undefined : idx;
   }
 
-  private _idxToInfo(idx: number): ExpandedTutorialStep | undefined {
-    const info = manifest.steps[idx];
-    if (info === undefined) {
-      return undefined;
-    }
-    const samplesRoot =
-      getCodeLanguagePreference() === 'ts' ? '/samples' : '/samples/js';
+  /**
+   * Generates a formatted object of data relevant to this tutorial for the
+   * given step index.
+   *
+   * @param idx Step index to load
+   * @returns A formatted info object of the given step. 404s if the step
+   *   does not exist in the file system
+   */
+  private _idxToInfo(idx: number): ExpandedTutorialStep {
+    const slug = `${idx}`.length == 1 ? `0${idx}` : `${idx}`;
+
+    const firstUrl = this.useOldUrl
+      ? '/tutorial/'
+      : `/tutorials/${this._projectLocation}`;
+
+    const nextUrl = this.useOldUrl
+      ? `/tutorial/#${slug}`
+      : `/tutorials/${this._projectLocation}#${slug}`;
```

Comment:
May need to add a trailing slash here between _projectLocation and slug. Although works either way so maybe doesn't matter.

Reply:
Will leave as so since every other url is written without the trailing `/`

Reply:
We should include it, because the canonical URL has a slash. If you click next, there won't be a slash, but then if you reload, it will appear (and there will be a 301 behind the scenes).